xbout/geometries.py

```diff
@@ -0,0 +1,119 @@
+from warnings import warn
+
+import xarray as xr
+import numpy as np
+
+REGISTERED_GEOMETRIES = {}
+
+
+class UnregisteredGeometryError(Exception):
+    """Error for unregistered geometry type"""
+
+
+def apply_geometry(ds, geometry_name):
+    """
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        Dataset (from
+    geometry_name : str
+        Name under which the desired geometry function was registered
+
+    Returns
+    -------
+    updated_ds
+
+    Raises
+    ------
+    UnregisteredGeometryError
+    """
+
+    try:
+        add_geometry_coords = REGISTERED_GEOMETRIES[geometry_name]
+    except KeyError:
+        raise UnregisteredGeometryError
```

Comment:
I think you can pass a string here to make the error message friendlier. As you've subclassed from `exception`, it should just do the right thing

Reply:
Good idea!